src/components/UserProfile/application/PendingApplications.tsx

```diff
@@ -0,0 +1,239 @@
+    challengeNameId: '',
+    opportunityNameId: '',
+  };
+
+  const { data: _opportunityNameId } = useOpportunityNameIdQuery({
+    variables: { ecoverseId: ecoverseID, opportunityId: opportunityID },
+    skip: !opportunityID,
+  });
+  if (opportunityID) {
+    nameIds.ecoverseNameId = _opportunityNameId?.ecoverse.nameID || '';
+    nameIds.challengeNameId = _opportunityNameId?.ecoverse.opportunity.challenge?.nameID || '';
+    nameIds.opportunityNameId = _opportunityNameId?.ecoverse.opportunity.nameID || '';
+  }
+
+  const { data: _challengeNameId } = useChallengeNameIdQuery({
+    variables: { ecoverseId: ecoverseID, challengeId: challengeID },
+    skip: !challengeID,
+  });
+  if (challengeID && !opportunityID) {
+    nameIds.ecoverseNameId = _challengeNameId?.ecoverse.nameID || '';
+    nameIds.challengeNameId = _challengeNameId?.ecoverse.challenge.nameID || '';
+  }
+
+  const { data: _ecoverseNameId } = useEcoverseNameIdQuery({
+    variables: { ecoverseId: ecoverseID },
+    skip: !!(ecoverseID && (challengeID || opportunityID)),
+  });
+  if (ecoverseID && !challengeID && !opportunityID) {
+    nameIds.ecoverseNameId = _ecoverseNameId?.ecoverse.nameID || '';
+  }
+
+  return (
+    <>
+      <Grid container spacing={1} justifyContent={'space-between'} alignItems={'center'} className={styles.row}>
+        <Grid item xs={6}>
+          <Link component={RouterLink} to={buildApplicationLink(nameIds, type)} aria-label="Link to entity">
+            <Typography className={styles.noPadding} noWrap={true} aria-label="Application display name">
+              {displayName}
+            </Typography>
+          </Link>
+        </Grid>
+        <Grid item xs={6} className={styles.labels}>
+          <Tag text={type} color="neutralMedium" aria-label="Application type" />
+          <Box display="flex" alignItems={'center'} className={styles.statusRow}>
+            <Tag
+              className={styles.capitalize}
+              text={state}
+              color={state === APPLICATION_STATE_NEW ? 'positive' : 'negative'}
+              aria-label="Application state"
+            />
+            <Tooltip title={t('tooltips.click-more-info')} placement="top">
+              <IconButton className={styles.noPadding} onClick={() => setShowDialog(true)} aria-label="Info dialog">
+                <InfoIcon />
+              </IconButton>
+            </Tooltip>
+            {edit && (
+              <Tooltip title={t('tooltips.click-delete')} placement="top">
+                <IconButton className={styles.noPadding} onClick={() => handleDelete(id)} aria-label="Delete">
+                  <DeleteIcon color={'error'} />
+                </IconButton>
+              </Tooltip>
+            )}
+          </Box>
+        </Grid>
+      </Grid>
+      <ApplicationDialogWrapper
+        onHide={() => setShowDialog(false)}
+        applicationId={id}
+        ecoverseId={ecoverseID}
+        show={showDialog}
+      />
+    </>
+  );
+};
```

Comment:
Separate presentation and business logic (Loading, actions etc.). Example: `OpportunityMembers`.

Another idea - same thing can be achieved by using hooks #1260 - `useApplicationCommunityQuery`

Can be discussed it more next week.